### PR TITLE
docs(version_guarantees): add minor may break

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

Minor version bumps can actually contain breaking changes, this lists in the guarantees that this may be the case.

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/version_guarantees.rst [hunk 1]
```diff
@@ -3,9 +3,25 @@
 Version Guarantees
 ==================
 
-The library follows a `semantic versioning principle <https://semver.org/>`_ which means that the major version is updated every time there is an incompatible API change. However due to the lack of guarantees on the Discord side when it comes to breaking changes along with the fairly dynamic nature of Python it can be hard to discern what can be considered a breaking change and what isn't.
+Nextcord does not follow semantic versioning exactly, The main difference is that breaking changes may also be made on minor version bumps (similar to Python's scheme).
```
Comment (Skelmis) [inline docs/version_guarantees.rst:6]: Lowercase the `t`

**Review by Skelmis — CHANGES_REQUESTED** [review]

**Review by Skelmis — APPROVED** [review]

**Review by EmreTech — APPROVED** [review]

**Comment by TAG-Epic** [comment]
Semver should be followed.

**Comment by EmreTech** [comment]
> Semver should be followed.

Okay then. Get Discord to somewhat follow Semver with their API, then we'll talk.

**Comment by ooliver1** [comment]
> Semver should be followed.

This change is so we are not limited to changing the major version every time there is a breaking change on Discord's API - which is pretty common. This reflects how discord.py actually handled such as [discord.py v1.5.0](https://discordpy.readthedocs.io/en/latest/whats_new.html#v1-5-0) when gateway intents were released.

**Review by DenverCoder1 — APPROVED** [review]
Sounds good to me.

(Using actual semver, the main repo would be on v4 before nc v3 is released 😃)

**Comment by ooliver1** [comment]
> > Sounds good to me.
> > (Using actual semver, the main repo would be on v4 before nc v3 is released 😃)
> 
> Yeah this has been a internal topic for a while now, I dont want to block NCv2 from bumping the major version due to bad naming.

I don’t know what internal conversations this is about, because I don't think it's involved anyone.

**Comment by ooliver1** [comment]
> Sounds good to me.
> 
> (Using actual semver, the main repo would be on v4 before nc v3 is released smiley)
> 
> **Updated with discussion from Discord**
> 
> If it's a breaking change that will affect very few users I don't think it needs a major release.
> 
> For something that many people would need to fix, a major would make sense.
> 
> One thing that's common for dependencies is a locked major version (eg. `nextcord>=2.0.0,<3`). That way the code is less likely to break from automatic dependency updates. So that's one reason to make all breaking changes major releases.
> 
> Some "breaking changes" are a bit of a stretch though and require making up an unlikely scenario, and for those, a major version change is not needed in my opinion.
> 
> Tbh it's not critical to go one way or the other as long as changes are clearly documented in the release notes.

Good words on the updates, we will go with that thank you. Time will see on what changes will require a major and what is a stretch.

**Pull request merged**